### PR TITLE
JDK-8301479: Replace NULL with nullptr in os/linux

### PR DESCRIPTION
**Pull request by jdksjolen**
Hi, this PR changes all occurrences of NULL to nullptr for the subdirectory os/linux. Unfortunately the script that does the change isn't perfect, and so we
need to comb through these manually to make sure nothing has gone wrong. I also review these changes but things slip past my eyes sometimes.

Here are some typical things to look out for:

1. No changes but copyright header changed (probably because I reverted some changes but forgot the copyright).
2. Macros having their NULL changed to nullptr, these are added to the script when I find them. They should be NULL.
3. nullptr in comments and logs. We try to use lower case "null" in these cases as it reads better. An exception is made when code expressions are in a comment.

An example of this:

```c++
// This function returns null
void* ret_null();
// This function returns true if *x == nullptr
bool is_nullptr(void** x);
```

Note how `nullptr` participates in a code expression here, we really are talking about the specific value `nullptr`.

Thanks!

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8301479](https://bugs.openjdk.org/browse/JDK-8301479): Replace NULL with nullptr in os/linux

### Reviewers
 * [Coleen Phillimore](https://openjdk.org/census#coleenp) (@coleenp - **Reviewer**)
 * [Severin Gehwolf](https://openjdk.org/census#sgehwolf) (@jerboaa - **Reviewer**) ⚠️ Review applies to [73e001d2](https://git.openjdk.org/jdk/pull/12315/files/73e001d230331b8f39d97c28ec0844e749e79e3b)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12315/head:pull/12315` \
`$ git checkout pull/12315`

Update a local copy of the PR: \
`$ git checkout pull/12315` \
`$ git pull https://git.openjdk.org/jdk pull/12315/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12315`

View PR using the GUI difftool: \
`$ git pr show -t 12315`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12315.diff">https://git.openjdk.org/jdk/pull/12315.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/os/linux/cgroupSubsystem_linux.hpp [hunk 1]
```diff
@@ -166,7 +166,7 @@ PRAGMA_DIAG_POP
   int err;                                                                \
   err = subsystem_file_line_contents(subsystem,                           \
                                      filename,                            \
-                                     NULL,                                \
+                                     nullptr,                                \
```
Comment (jdksjolen) [inline src/hotspot/os/linux/cgroupSubsystem_linux.hpp:169]: Align

File: src/hotspot/os/linux/cgroupSubsystem_linux.hpp [hunk 2]
```diff
@@ -184,11 +184,11 @@ PRAGMA_DIAG_POP
...
```
Comment (jdksjolen) [inline src/hotspot/os/linux/cgroupSubsystem_linux.hpp:187]: Align

File: src/hotspot/os/linux/cgroupSubsystem_linux.hpp [hunk 3]
```diff
@@ -184,11 +184,11 @@ PRAGMA_DIAG_POP
...
                                      scan_fmt,                            \
                                      variable);                           \
   if (err != 0)                                                           \
-    return (return_type) NULL;                                            \
+    return (return_type) nullptr;                                            \
```
Comment (jdksjolen) [inline src/hotspot/os/linux/cgroupSubsystem_linux.hpp:191]: Align

File: src/hotspot/os/linux/os_linux.cpp [hunk 4]
```diff
@@ -2167,14 +2167,14 @@ bool os::Linux::print_container_info(outputStream* st) {
...
+  st->print_cr("container_type: %s", p_ct != null ? p_ct : "not supported");
 
   char *p = OSContainer::cpu_cpuset_cpus();
-  st->print_cr("cpu_cpuset_cpus: %s", p != NULL ? p : "not supported");
+  st->print_cr("cpu_cpuset_cpus: %s", p != null ? p : "not supported");
   free(p);
 
   p = OSContainer::cpu_cpuset_memory_nodes();
-  st->print_cr("cpu_memory_nodes: %s", p != NULL ? p : "not supported");
+  st->print_cr("cpu_memory_nodes: %s", p != null ? p : "not supported");
```
Comment (jdksjolen) [inline src/hotspot/os/linux/os_linux.cpp:2177]: Glitched out

File: src/hotspot/os/linux/os_linux.cpp [hunk 5]
```diff
@@ -3305,29 +3305,29 @@ static char* anon_mmap(char* requested_addr, size_t bytes) {
   // succeed if we have enough swap space to back the physical page.
   char* addr = (char*)::mmap(requested_addr, bytes, PROT_NONE, flags, -1, 0);
 
-  return addr == MAP_FAILED ? NULL : addr;
+  return addr == MAP_FAILED ? nullptr : addr;
 }
 
 // Allocate (using mmap, NO_RESERVE, with small pages) at either a given request address
-//   (req_addr != NULL) or with a given alignment.
+//   (req_addr != null) or with a given alignment.
```
Comment (jdksjolen) [inline src/hotspot/os/linux/os_linux.cpp:3312]: nullptr

File: src/hotspot/os/linux/os_linux.cpp [hunk 6]
```diff
@@ -3815,36 +3815,36 @@ static char* shmat_with_alignment(int shmid, size_t bytes, size_t alignment) {
...
-  if (req_addr != NULL) {
+  if (req_addr != nullptr) {
     assert(is_aligned(req_addr, os::large_page_size()), "Must be divisible by the large page size");
     assert(is_aligned(req_addr, alignment), "Must be divisible by given alignment");
     return shmat_at_address(shmid, req_addr);
   }
 
   // Since shmid has been setup with SHM_HUGETLB, shmat will automatically
-  // return large page size aligned memory addresses when req_addr == NULL.
+  // return large page size aligned memory addresses when req_addr == null.
```
Comment (jdksjolen) [inline src/hotspot/os/linux/os_linux.cpp:3840]: nullptr

**Review by jdksjolen — COMMENTED** [review]
Found some things to fix

**Review by coleenp — APPROVED** [review]
Looks trivial.

**Review by jerboaa — APPROVED** [review]
Looks OK to me.

**Review by coleenp — APPROVED** [review]

**Comment by jdksjolen** [comment]
All tests passing, integrating. Thanks for review.

/integrate

**Pull request closed without merging**